shipstation/api.py

```diff
@@ -398,3 +417,37 @@ def post(self, endpoint='', data=None):
         )
         if self.debug:
             pprint.PrettyPrinter(indent=4).pprint(r.json())
+
+    def fetch_orders(self, parameters={}):
+        """
+            Query and fetch existing orders from ShipStation
+
+            Args:
+                parameters (dict): Dict of filters to filter by.
+
+            Returns: JSON string of the returned matching orders.
+
+            Raises:
+                AttributeError: parameters not of type dict
+                AttributeError: invalid key in parameters dict.
+
+            Examples:
+                >>> ss.fetch_orders(parameters={'order_status': 'shipped', 'page': '2'})
+        """
+
+        if not isinstance(parameters, dict):
+            raise AttributeError("`parameters` must be of type dict")
+
+        invalid_parameters = [
+            key for key in parameters.keys() if key not in self.ORDER_LIST_PARAMETERS
+        ]
+
+        if invalid_parameters:
```

Comment:
another optimization:
```suggestion
        invalid_keys = set(parameters.keys()).difference(self.ORDER_LIST_PARAMS)
        if invalid_keys:
            raise AttributeError("Invalid order list parameters: {}".format(
                ", ".join(invalid_keys)
            ))
```
You can drop the variable assignment/list comprehension above using this.

Additionally, `%` string interpolation is a faux pas, and `format` should be used instead.